src/util/cilfacade.ml

```diff
@@ -102,6 +122,7 @@ let createCFG (fileAST: file) =
   (* Since we want the output of justcil to compile, we do not run allBB visitor if justcil is enable, regardless of  *)
   (* exp.basic-blocks. This does not matter, as we will not run any analysis anyway, when justcil is enabled.         *)
   if not (get_bool "exp.basic-blocks") && not (get_bool "justcil") then end_basic_blocks fileAST;
+  if get_bool "ana.apron.threshold_widening" then compute_widening_thresholds fileAST;
```

Comment:
All of this logic could be performed inside the Apron lazy value itself. Then it doesn't have to be spread into seemingly unrelated modules.

Reply:
The reasoning might be that threshold widening could (should?) be a general concept, that we may later wish to apply also to our intervals...

Reply:
Sure, but then base intervals could also force the same lazy thing if they ever want it. It'd just have to be moved our of apron analysis for that to make more sense.

It's just cleaner if `Cilfacade` doesn't have to be cluttered with special-purpose things that all the analyses require and have to conditionally perform them in the right place during `createCFG`, which doesn't fit that function's purpose.